### PR TITLE
Update role/sites feature

### PR DESCRIPTION
**Pull request by mauriciofreshw**

### CODE REVIEW

File: client/src/constants/label.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const regionLabelsMap = {
+  region_interior: 'Interior',
+  region_fraser: 'Fraser',
+  region_vancouver_coastal: 'Vancouver Coastal',
+  region_vancouver_island: 'Vancouver Island',
+  region_northern: 'Northern',
+  none: 'None'
+};
```
Comment (Zenen-freshworks) [inline client/src/constants/label.js:8]: factoring stuff out of participantsview, nice!

File: cypress/integration/participantView.spec.js [hunk 2]
```diff
@@ -37,15 +37,15 @@ describe("Participant View", () => {
...
+    cy.get('ul.MuiMenu-list[role=listbox] > li').should('have.length', 7);
   });
 
   it("Visits Participant View as a MoH user", () => {
     cy.kcNavAs('ministry_of_health', 'participant-view');
     cy.get('div.MuiSelect-selectMenu').should('not.have.class', 'Mui-disabled');
     cy.get('div.MuiSelect-selectMenu').click();
     cy.get('ul.MuiMenu-list[role=listbox]').should('be.visible');
-    cy.get('ul.MuiMenu-list[role=listbox] > li').should('have.length', 6);
+    cy.get('ul.MuiMenu-list[role=listbox] > li').should('have.length', 7);
```
Comment (Zenen-freshworks) [inline cypress/integration/participantView.spec.js:48]: Question: if we have None as an option, should "Preferred Location" still be available as a catch-all? Do they serve different purposes?

**Review by Zenen-freshworks — CHANGES_REQUESTED** [review]
I'm getting an undefined error on filtering by None in ParticipantsView.

**Review by Zenen-freshworks — APPROVED** [review]

**Review by Zenen-freshworks — DISMISSED** [review]

**Review by Zenen-freshworks — APPROVED** [review]

File: client/src/pages/private/EOIView.js [hunk 3]
```diff
@@ -56,19 +56,10 @@ export default () => {
...
-      region_vancouver_island: 'Vancouver Island',
-      region_northern: 'Northern',
-      none: 'None'
-    };
-
     setHealthAuthorities(
       (roles.includes("superuser") || roles.includes("ministry_of_health"))
-      ? Object.values(locationRoles) 
-      : roles.map((loc) => locationRoles[loc]).filter(Boolean)
+        ? Object.values(regionLabelsMap).filter(value => value !== 'None')
```
Comment (fwkendall) [inline client/src/pages/private/EOIView.js:61]: I think this is incorrect, if I'm reading it right - MoH and superuser should have access to None for EEOIs

Comment (fwkendall) [reply client/src/pages/private/EOIView.js:61]: Note: does not apply to participants, None is only needed for EEOIs

File: server/server.js [hunk 4]
```diff
@@ -355,12 +361,43 @@ app.get(`${apiBaseUrl}/employer-sites-detail`,
     return res.json({ data: '' });
   }));
 
+app.post(`${apiBaseUrl}/update-user`,
```
Comment (fwkendall) [inline server/server.js:364]: should be patch :)

File: server/server.js [hunk 5]
```diff
@@ -355,12 +361,43 @@ app.get(`${apiBaseUrl}/employer-sites-detail`,
...
+        keycloakId: req.body.userId,
+      },
+      {
+        sites: req.body.sites,
+      },
+    );
+
+    const user = req.hcapUserInfo;
+    logger.info({
+      action: 'update-user_post',
```
Comment (fwkendall) [inline server/server.js:381]: just commenting so we remember to update this according to the endpoint update (user_patch)

File: server/server.js [hunk 6]
```diff
@@ -355,12 +361,43 @@ app.get(`${apiBaseUrl}/employer-sites-detail`,
...
+      performed_by: {
+        username: user.username,
+        id: user.id,
+      },
+      role_assigned: req.body.role,
+      granted_access_to: req.body.userId,
```
Comment (fwkendall) [inline server/server.js:387]: this key should probably have a different name for this type of action - user_id would be fine IMO (honestly I think it should have been like this for the other endpoint too, but that ship has sailed)

Comment (fwkendall) [reply server/server.js:387]: I guess we might want to leave it as-is, even though I think it's a bad key, so that it's consistent

**Review by ashtonmeuser — APPROVED** [review]

**Review by Zenen-freshworks — APPROVED** [review]

**Pull request merged**